server/src/main/java/org/elasticsearch/action/datastreams/GetDataStreamAction.java

```diff
@@ -307,6 +307,24 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params, @Nulla
                     builder.endArray();
                 }
                 builder.field(DataStream.GENERATION_FIELD.getPreferredName(), dataStream.getGeneration());
+                if (DataStream.isFailureStoreEnabled() && dataStream.getFailureStores().isEmpty() == false) {
```

Comment:
Should we return the `failure_stores` field as `"failure_stores": []` in the event that the failure store is enabled, but there are no backing indices, instead of checking for `.isEmpty()` here? It seems like we should just have the empty array if it's enabled even if no indices exist.